bgpd/bgp_route.c

```diff
@@ -11096,54 +11280,133 @@ static int bgp_table_stats(struct vty *vty, struct bgp *bgp, afi_t afi,
+				json_object_double_add(json,
+				     table_stats_strs[i][TABLE_STATS_IDX_JSON],
+				     ts.counts[i] ? (double)ts.counts[i] /
+				     (double)ts.counts[BGP_STATS_PREFIXES] : 0);
 			break;
 		case BGP_STATS_SPACE:
-			vty_out(vty, "%-30s: ", table_stats_strs[i]);
-			vty_out(vty, "%12g\n", ts.total_space);
-
+			if (!json) {
+				snprintf(temp_buf, sizeof(temp_buf), "%12g",
+					 ts.total_space);
+				vty_out(vty, "%-30s: %s\n",
+					table_stats_strs[i][TABLE_STATS_IDX_VTY],
+					temp_buf);
+			} else
+				json_object_double_add(json,
+				     table_stats_strs[i][TABLE_STATS_IDX_JSON],
+				     (double)ts.total_space);
 			if (afi == AFI_IP6) {
-				vty_out(vty, "%30s: ", "/32 equivalent ");
-				vty_out(vty, "%12g\n",
-					ts.total_space * pow(2.0, -128 + 32));
-				vty_out(vty, "%30s: ", "/48 equivalent ");
-				vty_out(vty, "%12g\n",
-					ts.total_space * pow(2.0, -128 + 48));
+				if (!json) {
+					snprintf(temp_buf, sizeof(temp_buf),
+						 "%12g", ts.total_space *
+						 pow(2.0, -128 + 32));
+					vty_out(vty, "%30s: %s\n",
+						"/32 equivalent %s\n",
+						temp_buf);
+				} else
+					json_object_double_add(json,
+						      "/32equivalent",
+						      (double)(ts.total_space *
+						      pow(2.0, -128 + 32)));
+				if (!json) {
+					snprintf(temp_buf, sizeof(temp_buf),
+						 "%12g", ts.total_space *
+						 pow(2.0, -128 + 48));
+					vty_out(vty, "%30s: %s\n",
+						"/48 equivalent %s\n",
+						temp_buf);
+				} else
+					json_object_double_add(json,
+						     "/48equivalent",
+						     (double)(ts.total_space *
+						     pow(2.0, -128 + 48)));
 			} else {
-				vty_out(vty, "%30s: ", "% announced ");
-				vty_out(vty, "%12.2f\n",
-					ts.total_space * 100. * pow(2.0, -32));
-				vty_out(vty, "%30s: ", "/8 equivalent ");
-				vty_out(vty, "%12.2f\n",
-					ts.total_space * pow(2.0, -32 + 8));
-				vty_out(vty, "%30s: ", "/24 equivalent ");
-				vty_out(vty, "%12.2f\n",
-					ts.total_space * pow(2.0, -32 + 24));
+				if (!json) {
+					snprintf(temp_buf, sizeof(temp_buf),
+						 "%12.2f",
+						 ts.total_space * 100. *
+						 pow(2.0, -32));
+					vty_out(vty, "%30s: %s\n",
+						"% announced ",
+						temp_buf);
+				} else
+					json_object_double_add(json,
+						     "%announced",
+						     (double)(ts.total_space *
+						     100. *
+						     pow(2.0, -32)));
+				if (!json) {
+					snprintf(temp_buf, sizeof(temp_buf),
+						 "%12.2f",
+						 ts.total_space *
+						 pow(2.0, -32 + 8));
+					vty_out(vty, "%30s: %s\n",
+						"/8 equivalent ", temp_buf);
+				} else
+					json_object_double_add(json,
+						    "/8equivalent",
+						    (double)(ts.total_space *
+						    pow(2.0, -32 + 8)));
+				if (!json) {
+					snprintf(temp_buf, sizeof(temp_buf),
+						 "%12.2f",
+						 ts.total_space *
+						 pow(2.0, -32 + 24));
+					vty_out(vty, "%30s: %s\n",
+						"/24 equivalent ",
+						temp_buf);
+				} else
+					json_object_double_add(json,
+					        "/24equivalent",
```

Comment:
just curious: do you want some json example with such field? :) interesting naming.

Reply:
if you have better name, I am interested.
please let me know